.github/actions/deploy/action.yaml

```diff
@@ -0,0 +1,182 @@
+name: Deploy
+inputs:
+  version:
+    required: true
+    type: string
+  base_url:
+    required: true
+    type: string
+  cloudflare_api_key:
+    required: true
+    type: string
+  database_url:
+    required: true
+    type: string
+  database_password:
+    required: true
+    type: string
+  mail_host:
+    required: true
+    type: string
+  mail_port:
+    required: true
+    type: string
+  mail_user:
+    required: true
+    type: string
+  mail_password:
+    required: true
+    type: string
+  redis_url:
+    required: true
+    type: string
+  strava_client_id:
+    required: true
+    type: string
+  strava_client_secret:
+    required: true
+    type: string
+  slack_token:
+    type: string
+  ssh_host:
+    required: true
+    type: string
+  ssh_key:
+    required: true
+    type: string
+  ssh_username:
+    required: true
+    type: string
+  token_secret:
+    required: true
+    type: string
+
+runs:
+  using: 'composite'
+  steps:
+    - name: Checkout
+      uses: actions/checkout@v4
+
+    - name: Add docker compose file to server
+      uses: appleboy/scp-action@v0.1.7
+      with:
+        host: ${{ inputs.SSH_HOST }}
+        username: ${{ inputs.SSH_USERNAME }}
+        key: ${{ inputs.SSH_KEY }}
+        source: docker/dev/docker-compose.yaml
+        target: trainer
+        strip_components: 2
+
+    - name: Start the containers with Docker Compose
+      uses: appleboy/ssh-action@v1.0.3
+      with:
+        host: ${{ inputs.SSH_HOST }}
+        username: ${{ inputs.SSH_USERNAME }}
+        key: ${{ inputs.SSH_KEY }}
+
+        script: |
+          cd trainer
```

Comment:
Still a lot of duplication in this step. I'll see if we can tighten this up in the future